source/common/filter/config_discovery_impl.h

```diff
@@ -606,6 +651,60 @@ class UpstreamHttpFilterConfigProviderManagerImpl
   const std::string getConfigDumpType() const override { return "ecds_filter_upstream_http"; }
 };
 
+// Network filter
+class NetworkFilterConfigProviderManagerImpl
+    : public FilterConfigProviderManagerImpl<
+          Server::Configuration::NamedNetworkFilterConfigFactory, NamedNetworkFilterFactoryCb,
+          Server::Configuration::FactoryContext,
+          NetworkDynamicFilterConfigProviderImpl<
+              Server::Configuration::FactoryContext,
+              Server::Configuration::NamedNetworkFilterConfigFactory>> {
+public:
+  absl::string_view statPrefix() const override { return "network_filter."; }
+
+protected:
+  bool
+  isTerminalFilter(Server::Configuration::NamedNetworkFilterConfigFactory* default_factory,
+                   Protobuf::Message& message,
+                   Server::Configuration::ServerFactoryContext& factory_context) const override {
+    return default_factory->isTerminalFilterByProto(message, factory_context);
+  }
+  void validateFilters(const std::string& filter_config_name, const std::string& filter_type,
+                       const std::string& filter_chain_type, bool is_terminal_filter,
+                       bool last_filter_in_filter_chain) const override {
+    Config::Utility::validateTerminalFilters(filter_config_name, filter_type, filter_chain_type,
+                                             is_terminal_filter, last_filter_in_filter_chain);
+  }
+  const std::string getConfigDumpType() const override { return "ecds_filter_network"; }
+};
+
+// Upstream network filter
+class UpstreamNetworkFilterConfigProviderManagerImpl
+    : public FilterConfigProviderManagerImpl<
+          Server::Configuration::NamedUpstreamNetworkFilterConfigFactory,
+          NamedNetworkFilterFactoryCb, Server::Configuration::CommonFactoryContext,
+          NetworkDynamicFilterConfigProviderImpl<
+              Server::Configuration::CommonFactoryContext,
+              Server::Configuration::NamedUpstreamNetworkFilterConfigFactory>> {
+public:
+  absl::string_view statPrefix() const override { return "network_filter."; }
```

Comment:
I think it's worth separating with a different prefix, maybe `upstream_network_filter`.

Reply:
For the HTTP version both downstream and upstream use the sane prefix so I thought I should follow the same. I can modify if you thinj otherwise, lmk.

Reply:
Separate is probably better. Less change of accidental collision. The upstream HTTP filter might have been an accident since they were added after ECDS.

Reply:
done